pyproject.toml

```diff
@@ -307,7 +307,7 @@ twine = "*"
 towncrier = ">=18.6.0rc1"
 
 [build-system]
-requires = ["poetry-core==1.2.0", "setuptools_rust==1.5.2"]
+requires = ["poetry-core>=1.0.0", "setuptools_rust>=1.3"]
```

Comment:
perhaps we *should* cap these to known-good versions, to avoid https://github.com/matrix-org/synapse/issues/13849 situations?

Reply:
No strong opinions here. I'd be happy to cap to `poetry-core>=1.0.0,<=1.3.0`, but I don't have any context on what's going on with `setuptools_rust` and how quickly-moving it is. E.g. I expect we'll want to take advantage of https://github.com/PyO3/setuptools-rust/pull/295 sooner rather than later.

Reply:
Please don't do `<=1.3.0`, version 1.3.1 is already out. If you're going to constrain the upper end of the version, at least make it `<1.4`.

Reply:
Discussed among the team. We'll constrain to `<= 1.3.1` and will raise it to 1.3.2/1.4.0/another version if a) someone wants to use that version, and b) we can check that it's safe.

Reply:
This is 045cecdb6.

Reply:
I think making a too-conservative upper bound conserves frustration by replacing "Synapse is broken" issues with "Requirements need updating" issues. Limiting to the current minor version might thread the needle by minimizing the likelihood of breakage while still allowing upgrades. (Of course, given how some Python projects seem to bump patch versions with ever merge and minor versions every third hour of every fourth day, maybe the relaxation isn't all that significant.)